### PR TITLE
Rename trait functions

### PR DESCRIPTION
**Pull request by gzanitti**
Since we also need name and type pairs for structs, it makes sense to rename `TraitFunction` to something more general (`NamedType` in this case).

There are also `NamedExpressions` and everything could be combined into a `NamedElement` or something similar that covers both cases. Personally I prefer this option, but the discussion is open.

### CODE REVIEW

File: parser/src/powdr.lalrpop [hunk 1]
```diff
@@ -719,20 +719,20 @@ PatternList: Vec<Pattern> = {
...
-    "trait" <name:Identifier> <type_vars:("<" <TraitVars> ">")> "{" <functions:TraitFunctions> "}" => TraitDeclaration { name, type_vars, functions }
+    "trait" <name:Identifier> <type_vars:("<" <TraitVars> ">")> "{" <functions:NamedTypes> "}" => TraitDeclaration { name, type_vars, functions }
 }
 
 TraitVars: Vec<String> = {
     <mut list:( <TypeVar> "," )*> <end:TypeVar> ","?  => { list.push(end); list }
 }
 
-TraitFunctions: Vec<TraitFunction<Expression>> = {
+NamedTypes: Vec<NamedType<Expression>> = {
```
Comment (chriseth) [inline parser/src/powdr.lalrpop:729]: Can you move this to a more generic position?

**Review by chriseth — COMMENTED** [review]
Looks good apart from the organization of the grammar file.

**Comment by gzanitti** [comment]
I am thinking that we may also have to change the definition of the `NamedType` in the grammar, since in `Traits` we needed functions but now it is no longer the case.

```
NamedType: NamedType<Expression> = {
    <name:Identifier> ":" <params:TypeTermList<Expression>> "->" <value:TypeTermBox<Expression>> => NamedType { name, ty: Type::Function(FunctionType{params, value}) }
}
```

**Comment by chriseth** [comment]
Then let's keep the two different grammar rules but combine the rust types?

**Comment by chriseth** [comment]
Oh and we certainly have to allow functions, but we can just use `AStruct { a: (int -> int), b: (AStruct -> bool) }`, right?

**Review by chriseth — APPROVED** [review]

**Pull request merged**